hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -95,7 +97,7 @@ public EnhancerImpl(EnhancementContext enhancementContext) {
 		try {
 			final TypeDescription managedCtClass = classPool.describe( className ).resolve();
 			DynamicType.Builder<?> builder = doEnhance(
-					new ByteBuddy().with( TypeValidation.DISABLED ).redefine( managedCtClass, ClassFileLocator.Simple.of( className, originalBytes ) ),
+					new ByteBuddy().ignore( isDefaultFinalizer() ).with( TypeValidation.DISABLED ).redefine( managedCtClass, ClassFileLocator.Simple.of( className, originalBytes ) ),
```

Comment:
By default, ByteBuddy ignores synthetic methods and the default finalizer but since there are several tests with inner classes it's best to enhance those synthetic methods than change the tests. Also, for consistency with Javassist.